libs/store/src/lib/reducers/add.ts

```diff
@@ -0,0 +1,83 @@
+import { AppStateModel, AppTournamentId } from '@razor/models';
+import { extractId, extractIdType } from '@razor/util';
+import {
+  addPlayerReducerPayload,
+  addRaceReducerPayload,
+  addTournamentReducerPayload,
+} from '../payloads';
+
+// Basic Add Operations
+export const addTournamentReducer = (
+  state: AppStateModel,
+  payload: addTournamentReducerPayload,
+): AppStateModel => {
+  const { tournamentId, tournament } = payload;
+  const newState: AppStateModel = {
+    ...state,
+    tournamentsModel: {
+      ...state.tournamentsModel,
+      [tournamentId]: tournament,
+    },
+  };
+  return newState;
+};
+
+export const addRaceReducer = (
+  state: AppStateModel,
+  payload: addRaceReducerPayload,
+): AppStateModel => {
+  const { raceId, race } = payload;
+  const tournamentId: AppTournamentId = extractId(
+    raceId,
+    extractIdType.race,
+    extractIdType.tournament,
+  ) as AppTournamentId;
+
+  if (raceId) {
```

Comment:
Do we expect `raceId` to be undefined?

Reply:
Fixed in PR #14 under commit 39ef350e9fa663424ef9b78c7f25a8c240263c36